bin/sl-meshctl.js

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/env node
+/* eslint no-console:0 no-process-exit:0 */
+
+var Client = require('../client/client');
+var Parser = require('posix-getopt').BasicParser;
+var _ = require('lodash');
+var assert = require('assert');
+var concat = require('concat-stream');
+var debug = require('debug')('strong-mesh-client:meshctl');
+var fs = require('fs');
+var home = require('osenv').home();
+var npmls = require('strong-npm-ls');
+var path = require('path');
+var sprintf = require('sprintf');
+var url = require('url');
+var userHome = require('user-home');
+var util = require('util');
+var maybeTunnel = require('strong-tunnel');
+
+assert(userHome, 'User home directory cannot be determined!');
+
+function printHelp($0, prn) {
+  var USAGE = fs.readFileSync(require.resolve('./sl-meshctl.txt'), 'utf-8')
+    .replace(/%MAIN%/g, $0)
+    .trim();
+
+  prn(USAGE);
+}
+
+var argv = process.argv;
+var $0 = process.env.CMD || path.basename(argv[1]);
+var parser = new Parser([
+  ':v(version)',
+  'h(help)',
+  'C:(control)'
+].join(''), argv);
+
+var apiUrl = process.env.STRONGLOOP_MESH_API ||
+  process.env.STRONGLOOP_PM ||
+  exists('pmctl') ||
+  exists(path.join(home, '.strong-pm', 'pmctl')) ||
+  '/var/lib/strong-pm/pmctl' ||
+  'http://127.0.0.1:8701';
+
+var command = 'status';
+var option;
+while ((option = parser.getopt()) !== undefined) {
+  switch (option.option) {
+    case 'v':
+      console.log(require('../package.json').version);
+      process.exit(0);
+      break;
+    case 'h':
+      printHelp($0, console.log);
+      process.exit(0);
+      break;
+    case 'C':
+      apiUrl = option.optarg;
+      break;
+    default:
+      console.error('Invalid usage (near option \'%s\'), try `%s --help`.',
+        option.optopt,
+        $0);
+      process.exit(1);
+      break;
+  }
+}
+
+var optind = parser.optind();
+
+if (optind < argv.length) {
+  command = argv[optind++];
+}
+
+debug('API Url: %s', apiUrl);
+
+if (!command) {
+  console.error('No command, try `%s --help`.', $0);
```

Comment:
I see. I think this code isn't reachable: command starts with a value, it can only change by getting a new value. You might be able to hit this block by doing `meshctl 0` (a command of `0`, falsy), but in that case the msg is weird, and would be better handled down below, by the `unknown()` command handler.